### DISCUSSION TITLE
2023 KAIST SoC Colloquium 질문과 답변

### DISCUSSION THREAD

**Discussion by orioncactus** [orioncactus/pretendard#135] [General]
# 개요
2023년 6월 5일 월요일 오후 4시에 [KAIST](https://cs.kaist.ac.kr/colloquium/info.php?id=12)에서 진행하는 강연에 대한 질문과 답변을 나눕니다.

# 질문 올리기
아래 댓글로 편하게 질문을 올려주세요.
좋은 질문이 보인다면, 댓글에 있는 **`↑`** 버튼으로 우선 순위를 높여주시면 우선 순위에 맞게 답변을 드리겠습니다 (▰˘◡˘▰)

**Comment by minsukk-at-kaist-ac-kr** [comment]
다양한 환경, 다른 독자를 대상으로 어떻게 하면 가장 좋은 폰트를 선택할 수 있을까요? 

예를 들어 50대 공무원을 대상으로 하는 문서와, 10대 대학생을 대상으로하는 문서에 사용할 폰트는 다르게 사용될 때 그 효과가 극대화 될것 같다는 생각이 듭니다. 이럴 때 나와는 다른 예상 독자가 어떤 폰트를 더 좋은 폰트라고 생각할 지 어떻게 미리 알 수 있을까요?

**Reply by orioncactus** [reply]
사실 저는 잘못된 폰트는 없고, 또 좋은 폰트는 굴림 같은 어느 폰트도 될 수 있다고 생각해요. 그래서 50대 공무원이나 10대나 대학생이 특정 폰트를 쓰는 목적을 먼저 살펴보고, 그 목적에 부합하는 폰트를 비교해서 선택하면 고민을 풀 수 있을 것 같아요. 

그리고 폰트를 통해 10대나 50대가 가진 환경에 각각 맞춰야 하는 이유를 생각해보면, 각 세대가 가진 환경적 문제를 해결하는 쪽으로 볼 수 있을 것 같은데요, 그렇다면 50대에 맞춰진 폰트는 신체 기능 저하로 특수한 상황에서 봐도 잘 보일 수 있는 형태를 가진 폰트를 찾아볼 수 있겠고, 10대나 대학생은 신체적인 문제보다는 시간적인 문제로 같은 공간에 더 빠르게 읽을 수 있는 폰트를 찾아볼 수 있을 듯합니다.

또 50대 공무원이 선호하는 폰트 얘기를 실제로 들어보면, 대개 그 폰트를 예전부터 썼기 때문에 정석처럼 인지했기 때문이거나, 또는 너무나 바꾸고 싶으나 그 폰트를 바꾸기 위한 인프라 제고가 어렵다거나, 또는 담당 부서가 구성되지 않았다거나 하는 이유더라고요. 그래서 초점을 폰트 자체보다는 폰트를 통해 얻고자 하는 가치가 무엇인지로 확장해서 파악해보면 좋을 것 같습니다.

만약 특정 폰트를 쓰는 특별한 이유가 있는 게 아니면, 전세계적으로 폰트를 대통합했던 산세리프 폰트인 Helvetica와 비슷한 결을 가진 본문 폰트를 쓰면 웬만한 경우는 포용할 수 있으니, 해당 결을 가진 Pretendard를 써도 좋지 않을까 싶습니다 😊

**Comment by actruce** [comment]
인터넷에 보면 많은 무료폰트들이 있는데
좋은 폰트를 고르는 것은 생각보다 쉽지 않더라구요
좋은 폰트를 고를 수 있는 팁이 있을까요?

**Reply by orioncactus** [reply]
좋은 폰트라고 생각할 수 있는 기준은 여러가질 수 있는데요, 저는 크게 기능적인 부분에 제약이 없는 폰트와, 폰트가 표방하는 목적에 부합하는 모양을 가진 폰트로 그 기준을 나눠서 구분하는 편이에요.

전자는 서비스나 제품에 쓰였을 때 더 빠르게 목적을 이룰 수 있는 폰트로 생각하고, 예를 들면 개발과 디자인에서 썼을 때 불편함이 없는 폰트거나, 폰트에 다양한 기능들로 여러 상황에 대응할 수 있는 폰트거나 하는 부분으로 살펴볼 수 있을 것 같고요, 후자는 형태에 관한 부분으로, 다양한 폰트 스타일 중에 특정 부분의 표현이 뛰어나거나, 아쉬운 문자 모양이 없거나 하는 부분으로 살펴볼 수 있을 것 같아요.

만약에 이런 기준을 생각해보기에 어려움이 있다면, 단순히 사람들이 많이 찾는 폰트를 골라서 흐름에 편입해도 괜찮을 것 같아요. 무료 폰트를 모아볼 수 있는 [눈누](https://noonnu.cc/)에는 모든 폰트 탭에 인기순이나 조회순으로 폰트를 골라볼 수 있고, [Google Fonts](https://fonts.google.com/)에도 마찬가지로 트렌딩이나 가장 많이 쓰인 순으로 폰트를 골라볼 수 있는데, 이런 폰트들을 한번 검색해봐서 이 폰트는 어떨 때 제 역할을 발휘하는지 살펴보면 좋은 무료 폰트를 잘 고르는 데 도움이 될 것 같습니다.

**Comment by Lim-Sangho** [comment]
Pretendard라는 이름을 어떻게 짓게 되었는지 궁금합니다.

**Reply by orioncactus** [reply]
Pretendard는 완전히 새롭게 만들어진 폰트이기보다는, 기존 폰트를 살짝 개선해 범용성을 높인 폰트로 개념을 잡아두었어요. 그래서 제가 이 폰트를 만들면서 기준으로 두었던 Apple 시스템 폰트를 바탕으로 표준과 근접한 폰트를 만들었으니, 애초에 이름도 대놓고 그런 폰트다, 라고 말하면 어떨까 싶어 생각해보고, 표준이라는 영어 단어인 Standard에 근접한 게 뭘까 생각하다가, 가장한다는 뜻을 가진 Pretend를 붙이면 자연스럽겠더라고요. 그래서 Pretend와 Standard를 붙여 Pretendard라는 이름으로 지었어요.

또 여러 검색 서비스에 검색해봤을 때 Pretendard가 거의 없던 단어기도 했어서, 고유명사로 불리기 충분할 듯한 점도 있었습니다 😁

추가적으로는 한국어로 이름을 지어도 되긴 했지만, 저는 가능하면 이 폰트가 국제적으로 쓰여도 문제 없을 것으로 생각해서, 좀 더 범용성 있게 이름을 가져가고자 했어요. 그래서 한국어에서 생각해보면 발음하기는 어렵긴 하지만, 프리텐다드로 부르고 보여져도 괜찮을 것 같다 싶은 것도 있었어요.

**Comment by actruce** [comment]
상세한 답변 감사드립니다
좋은 하루 가득한 휴일 되세요

2023년 6월 6일 (화) 02:08, Kil Hyung-jin ***@***.***>님이 작성:

**Comment by slam64** [comment]
안녕하세요. 좋은 폰트 만들어 주셔서 감사합니다!
덕분에 회사에서 유용하게 사용하고 있습니다^^
최근 1.3.7 버전으로 업데이트가 되었는데요... 
이번 버전부터  dynamic subset font가 압축 파일에서 제외한다고 쓰여 있던데
제외한 이유가 있을까요?

**Reply by orioncactus** [reply]
정확히는 릴리즈에서 제공하는 zip 파일에서, 기존에 제외했던 다이나믹 서브셋과 같이 이번 릴리즈에서 가변 다이나믹 서브셋도 제외한 것입니다. 다이나믹 서브셋은 보통 CDN URL로 사용하시고, 필요한 경우에만 다운로드해서 자체 서버에서 사용하시기 때문에, 일반적인 목적으로 쓰는 zip로 제공할 때 다이나믹 서브셋 파일로 길어지는 압축 해제 시간만큼 쓰임이 있을 것 같지 않다고 판단했습니다.

혹시 zip 파일에서 다이나믹 서브셋이 필요하시다면 이유를 알 수 있을까요? 다이나믹 서브셋을 zip에 포함하지 않았으나, 필요하시다면 레포지터리를 다운로드해서 사용하시는 방법이 있습니다.

**Reply by slam64** [reply]
아 먼저 답변 감사합니다!

> 혹시 zip 파일에서 다이나믹 서브셋이 필요하시다면 이유를 알 수 있을까요?
이 답을 드리자면 cdn이 해외 도메인이라 혹시 망이 차단되는 사태가 발생하면 폰트 적용이 
안 되는 경우도 발생하지 않을까 하는 개발자 요청이 있어서요- 제가 보기에 가능성이 적어 보이긴 했지만
그렇다고 100% 문제 없다고 확신이 없더군요... 그래서 저희 쪽 서버 내부망에 설치를 하려고 했는데
zip 파일에 다이나믹 서브셋 파일이 없다는 걸 알게 되어서 그랬습니다... (정확히 말하면 이번 버전에
가변 다이나믹 서브셋이 제외 된 건데 저는 일반 다이나믹 서브셋이 제외되었다고
착각했습니다만 아무튼 그건 논외로 치고^^;;)

말씀하신 대로 저장소에서 다운 받으면 되겠네요!

다시 한번 감사드립니다!

**Reply by orioncactus** [reply]
jsDelivr의 안정성에 대해서는 크게 염려하지 않아도 괜찮지 않을까 싶습니다. [jsDelivr 웹사이트 설명](https://www.jsdelivr.com/network)에 따르면, jsDelivr는 Cloudflare, Fastly, Bunny, Gcore 중에서 가장 빠른 CDN 네트워크를 사용하며, 이 중 네 개 모두 장애가 발생할 때 Amazon S3, 그리고 이마저도 장애가 생길 때는 GitHub에서 직접 파일을 가져온다고 합니다. 또 jsDelivr는 유명한 Vue와 Bootstrap에서 공식 CDN으로 사용하고 있기도 해서, jsDelivr로 쓰셔도 무방하다고 안내드리기는 하는데요, 말씀하신 것처럼 직접 서빙하고자 하신다면 저장소를 통해 쓰시는 것을 추천드리며, 개발자 분께 설득이 필요하시다면 위 내용과 함께 이전에 관련해서 나눈 이슈 #51 도 참고 겸 안내드려도 좋을 듯합니다 😌

**Reply by slam64** [reply]
오! 질문하길 잘 했네요~ 친절한 답변 감사합니다 :)
